### ISSUE TITLE
Size of the dataset for finetuning

### ISSUE THREAD

**Issue by bkhanal-11** [meta-llama/llama-recipes#420]
Thank you for your amazing work and help with other issues as well.

I was curious about how large should the dataset be if I want to fine-tune on llama2-7b. Like if I want to train on a QA dataset, how many samples need to be there for successful fine-tuning? I tried to test this with just 267 pairs of questions and answers on a custom dataset with a modified [alpaca_dataset.py](https://github.com/meta-llama/llama-recipes/blob/main/src/llama_recipes/datasets/alpaca_dataset.py). The script I ran for fine-tuning was

```bash
finetuning.py --use_peft --peft_method lora --quantization --use_fp16 --model_name ../../../llama/llama-2-7b-hf/ --output_dir finetuned_model/
```

and for inference was

```bash
python3 inference.py --model_name ../../../../llama/llama-2-7b-hf/ --peft_model ../../finetuning/finetuned_model/ --prompt_file custom_prompt.txt
```

The response from LLM was far from perfect. What am I missing here? I am open to any suggestions and help. Let me know if you need further information. Thank you in advance.

**Comment by HamidShojanazeri** [comment]
@bkhanal-11 thanks for your interest, can you please elaborate on the goal a bit? are you trying to fine-tune a base model to be a chat /assistant model? I am reading this from using alpaca dataset. Assuming this is for experimentation but general though there, Alpaca was around 52K and [OpenAssistant around 80 k](https://github.com/meta-llama/llama-recipes/blob/main/recipes/finetuning/datasets/custom_dataset.py), that kinda signals the ball park of required examples. Also, if you intent to adapt to a new domain, it might be helpful to start with a chat model `llama-2-7b-chat-hf`.

**Comment by bkhanal-11** [comment]
Hi @HamidShojanazeri thank you for your response. I am trying to finetune for a domain-specific QA task. I have question-answer pairs related to FAQ for a web service. Since I have limited samples, I was not getting the expected response after fine-tuning. You mentioned `llama-2-7b-chat-hf`, is there any advantage of using this instead of base `llama-2-7b-hf`?

Also, this is not related to this issue, but you might be able to help me. Is there any way I can load the models directly using PyTorch? Right now, as far as I understand, we have to convert the provided model to Hugginface format. I was wondering if that is necessary. Is there any way to use the weights directly with PyTorch

**Comment by NoviceStone** [comment]
Hi @bkhanal-11, to load the weights directly using PyTorch, you can specify the `use_safetensors` argument in the script `llama_recipes/finetuning.py`.

```python
        model = LlamaForCausalLM.from_pretrained(
            train_config.model_name,
            load_in_8bit=True if train_config.quantization else None,
            device_map="auto" if train_config.quantization else None,
            use_cache=use_cache, 
            attn_implementation="sdpa" if train_config.use_fast_kernels else None,
            use_safetensors=False,
        )
```

**Comment by HamidShojanazeri** [comment]
@bkhanal-11 The chat model is already instruction fine-tuned so you wont need to make the based model instruction fine-tuned anymore to act as an assistant. 

For the second question, as @NoviceStone mentioned, there is no need for conversion you can use HF checkpoints directly.

**Comment by bkhanal-11** [comment]
Thank you for the information @HamidShojanazeri and @NoviceStone.
